src/ray/raylet/lineage_cache_test.cc

```diff
@@ -180,13 +179,36 @@ TEST_F(LineageCacheTest, TestGetUncommittedLineage) {
   }
 
   // Get the uncommitted lineage for the inserted task.
-  uncommitted_lineage = lineage_cache_.GetUncommittedLineage(combined_task_ids.back());
+  uncommitted_lineage =
+          lineage_cache_.GetUncommittedLineage(combined_task_ids.back(), ClientID::nil());
   // Check that the uncommitted lineage is exactly equal to the entire set of
   // tasks inserted so far.
   ASSERT_EQ(combined_task_ids.size(), uncommitted_lineage.GetEntries().size());
   for (auto &task_id : combined_task_ids) {
     ASSERT_TRUE(uncommitted_lineage.GetEntry(task_id));
   }
+
+}
+
+TEST_F(LineageCacheTest, TestMarkTaskAsForwarded) {
+  // Insert chain of tasks.
+  std::vector<Task> tasks;
+  auto return_values =
+          InsertTaskChain(lineage_cache_, tasks, 4, std::vector<ObjectID>(), 1);
+  std::vector<TaskID> task_ids;
+  for (const auto &task : tasks) {
+    task_ids.push_back(task.GetTaskSpecification().TaskId());
+  }
+
+  auto node_id = ClientID::from_random();
+  auto forwarded_task_id = task_ids[task_ids.size() - 2];
+  auto remaining_task_id = task_ids[task_ids.size() - 1];
+  lineage_cache_.MarkTaskAsForwarded(forwarded_task_id, node_id);
+  auto uncommitted_lineage =
+          lineage_cache_.GetUncommittedLineage(remaining_task_id, node_id);
+
+  ASSERT_EQ(1, uncommitted_lineage.GetEntries().size());
+  ASSERT_TRUE(uncommitted_lineage.GetEntry(remaining_task_id));
```

Comment:
I would also add an assertion that the uncommitted lineage for a random other node ID is still the full size.